src/main/java/com/aws/greengrass/secretmanager/SecretManager.java

```diff
@@ -353,6 +356,29 @@ private GetSecretValueResponse getSecret(String secretId, String versionId, Stri
         return cache.get(arn + LATEST_LABEL);
     }
 
+    private GetSecretValueResponse getSecret(String secretId, String versionId, String versionStage,
+                                             boolean refreshSecret) throws GetSecretException {
+        logger.atDebug().kv("secretId", secretId).kv("versionId", versionId).kv("versionStage", versionStage)
+                .log("get-secret");
+        String arn = validateSecretId(secretId);
+
+        // Both are optional
+        if (!Utils.isEmpty(versionId) && !Utils.isEmpty(versionStage)) {
+            logger.atError().kv("secretId", secretId).kv("versionId", versionId).kv("versionStage", versionStage)
+                    .log("Both secret version and id are set");
+            throw new GetSecretException(400, "Both versionId and Stage are set in the request");
+        }
+        /*
+         If refresh is set to true, try fetching the secret from cloud. This will update the local store and cache as
+          well. Refresh secrets by label only. If refreshing the secret fails for any reason, we fall back to local
+          store.
+         */
+        if (refreshSecret && Utils.isEmpty(versionId)) {
```

Comment:
when triggered via IPC this should not run with any retries, only the default retries in the AWS SDK.

Reply:
Ok, but we can still send the existing one from cache and have this retried in the background if we make this non-blocking, right? In any case, I'll update those in the later PRs.

Reply:
It should not retry forever. It may retry 3 times as is the default in the AWS SDK